### PR TITLE
Add ISLCoreErrorLogger

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Part of #5299

### CODE REVIEW

File: src/SLCore/ISLCoreErrorLoggerFactory.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            while (!cancellationTokenSource.IsCancellationRequested)
+            {
+                var line = await errorStream.ReadLineAsync();
+                if (string.IsNullOrEmpty(line))
+                {
+                    continue;
+                }
+
+                logger.LogVerbose("[SLCORE-ERR]" + line);
```
Comment (pavel-mikula-sonarsource) [inline src/SLCore/ISLCoreErrorLoggerFactory.cs:91]: `continue` should be avoided where possible, it makes the logic hard to follow
```suggestion
                if (!string.IsNullOrEmpty(line))
                {
                    logger.LogVerbose("[SLCORE-ERR]" + line);
                }
```

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore/ISLCoreErrorLoggerFactory.cs]: I disagree. It is the case when you have many statements inside of the If body and there's a lot of branching, but here it actually makes the intent of skipping the iteration explicit. if I saw the suggested variant, I would think that someone forgot to finish the method. The good alternative would be to move the IsNullOrEmpty condition to the while statement, but it won't look great in this particular case due to variable assignment.

Comment (ugras-ergun-sonarsource) [reply src/SLCore/ISLCoreErrorLoggerFactory.cs]: I agree that in this case having "continue" makes it more obvious of the intentionality of the skipping.

**Review by ugras-ergun-sonarsource — APPROVED** [review]
